bundles/org.openhab.binding.solarman/src/main/java/org/openhab/binding/solarman/internal/modbus/SolarmanV5Protocol.java

```diff
@@ -0,0 +1,270 @@
+                        + offsetTime.length + requestFrame.length)
+                .put(frameType).put(sensorType).put(totalWorkingTime).put(powerOnTime).put(offsetTime).put(requestFrame)
+                .array();
+    }
+
+    /**
+     * Based on <a href="https://www.modbustools.com/modbus.html#function03">Function 03 (03hex) Read Holding
+     * Registers</a>
+     *
+     * @param slaveId - Slave Address
+     * @param mbFunctionCode -
+     * @param firstReg - Starting Address
+     * @param lastReg - Ending Address
+     * @return byte array containing the Modbus request frame
+     */
+    protected byte[] buildModbusReadHoldingRegistersRequestFrame(byte slaveId, byte mbFunctionCode, int firstReg,
+            int lastReg) {
+        int regCount = lastReg - firstReg + 1;
+        byte[] req = ByteBuffer.allocate(6).put(slaveId).put(mbFunctionCode).putShort((short) firstReg)
+                .putShort((short) regCount).array();
+        byte[] crc = ByteBuffer.allocate(Short.BYTES).order(ByteOrder.LITTLE_ENDIAN)
+                .putShort((short) CRC16Modbus.calculate(req)).array();
+
+        return ByteBuffer.allocate(req.length + crc.length).put(req).put(crc).array();
+    }
+
+    protected Map<Integer, byte[]> parseModbusReadHoldingRegistersResponse(byte @Nullable [] frame, int firstReg,
+            int lastReg, Boolean allowLogging) {
+        int regCount = lastReg - firstReg + 1;
+        Map<Integer, byte[]> registers = new HashMap<>();
+        int expectedFrameDataLen = 2 + 1 + regCount * 2;
+        if (frame == null || frame.length < expectedFrameDataLen + 2) {
+            if (allowLogging) {
+                logger.error("Modbus frame is too short or empty");
+            }
+
+            return registers;
+        }
+
+        int actualCrc = ByteBuffer.wrap(frame, expectedFrameDataLen, 2).order(ByteOrder.LITTLE_ENDIAN).getShort()
+                & 0xFFFF;
+        int expectedCrc = CRC16Modbus.calculate(Arrays.copyOfRange(frame, 0, expectedFrameDataLen));
+
+        if (actualCrc != expectedCrc) {
+            if (allowLogging) {
+                logger.error("Modbus frame crc is not valid. Expected {}, got {}", String.format("%04x", expectedCrc),
+                        String.format("%04x", actualCrc));
+            }
+
+            return registers;
+        }
+
+        for (int i = 0; i < regCount; i++) {
+            int p1 = 3 + (i * 2);
+            ByteBuffer order = ByteBuffer.wrap(frame, p1, 2).order(ByteOrder.BIG_ENDIAN);
+            byte[] array = new byte[] { order.get(), order.get() };
+            registers.put(i + firstReg, array);
+        }
+
+        return registers;
+    }
+
+    protected byte @Nullable [] extractModbusResponseFrame(byte @Nullable [] responseFrame, byte[] requestFrame,
+            Boolean allowLogging) {
+        if (responseFrame == null || responseFrame.length == 0) {
+            if (allowLogging) {
```

Comment:
I see this allowLogging everywhere, i guess this is some debugging method, please remove and use the default logging options provided by the framework.

Reply:
Removed allowLogging .